app/javascript/components/AddFlight.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import {
+  postFlight, setPicture, setAvailableSlots, setBasePrice, setName,
+} from '../redux/flights/flightsSlice';
+
+const AddFlight = () => {
+  const dispatch = useDispatch();
+  const history = useNavigate();
+
+  const {
+    name, picture, basePrice, availableSlots,
+    userId,
+  } = useSelector((store) => store.flights);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    const formData = new FormData();
+    formData.append('flight[name]', name);
+    formData.append('flight[picture]', picture);
+    formData.append('flight[base_price]', basePrice);
+    formData.append('flight[available_slots]', availableSlots);
+    formData.append('flight[reserved]', 'false');
+    formData.append('flight[user_id]', userId);
+
+    dispatch(postFlight(formData));
+    history('/');
+  };
+
+  return (
+
+    <div className="modal fade" id="addModal" tabIndex="-1" aria-labelledby="addModalLabel" aria-hidden="true">
+      <form onSubmit={handleSubmit}>
+        <div className="modal-dialog">
+          <div className="modal-content">
+            <div className="modal-header">
+              <h1 className="modal-title" id="addModalLabel">Add a New Flight</h1>
+              <button type="button" className="btn-close" data-bs-dismiss="modal" aria-label="Close" />
+            </div>
+            <div className="modal-body">
+              <input
+                className="form-control mt-3"
+                type="text"
+                value={name}
+                placeholder="Flight Name"
+                onChange={(e) => dispatch(setName(e.target.value))}
+                required
+                data-testid="flight-name-input"
+
+              />
```

Comment:
[OPTIONAL] this input is repeated a couple of times, maybe it's a good candidate to make into a subcomponent and make the code a bit cleaner?

Reply:
Hello @mcrd25, Thank you so much for the meticulous and painstaking work done to review our Capstone Project. Concerning this optional comment you suggested on AddFLight.js component, when I tried to implement it, I noticed it still boils down to the same lengthy episode because each input field has a separate value, placeholder, dispatched action....


Please @mcrd25 do you think there is a better way to accomplish this?

The snippet is attached below:

The extracted component:

```
import React from 'react';

const CustomInput = ({ value, placeholder, onChange, dataTestId }) => (
  <input
    className="form-control mt-3"
    type="text"
    value={value}
    placeholder={placeholder}
    onChange={onChange}
    required
    data-testid={dataTestId}
  />
);

export default CustomInput;

```


Then the application of the subcomponent which, in my own opinion, is not cleaner....
```
<div className="modal-body">
              <CustomInput
                value={name}
                placeholder="Flight Name"
                onChange={(e) => dispatch(setName(e.target.value))}
                dataTestId="flight-name-input"
              />

              <CustomInput
                value={picture}
                placeholder="Picture URL for the flight"
                onChange={(e) => dispatch(setPicture(e.target.value))}
                dataTestId="flight-picture-input"
              />

              <CustomInput
                value={basePrice}
                placeholder="Base Price for the destination"
                onChange={(e) => dispatch(setBasePrice(e.target.value))}
                dataTestId="flight-base-price-input"
              />

              <CustomInput
                value={availableSlots}
                placeholder="Available Slots for the destination"
                onChange={(e) => dispatch(setAvailableSlots(e.target.value))}
                dataTestId="flight-available-slots-input"
              />
            </div>
```

Please let us know if you have a better way to implement what you're suggesting.